vdr/validators.go

```diff
@@ -38,14 +40,24 @@ import (
 //   - every service id must have a fragment
 //   - every service id should have the DID prefix
 //   - every service id must be unique
-func CreateDocumentValidator() did.Validator {
+//   - every service type must be unique
+func NetworkDocumentValidator() did.Validator {
 	return &did.MultiValidator{Validators: []did.Validator{
 		did.W3CSpecValidator{},
 		verificationMethodValidator{},
-		serviceValidator{},
+		basicServiceValidator{},
 	}}
 }
 
+// ManagedDocumentValidator extends NetworkDocumentValidator with extra safety checks to be performed on DID documents managed by this node before they are published on the network.
+func ManagedDocumentValidator(serviceResolver didservice.ServiceResolver) did.Validator {
+	validator := NetworkDocumentValidator().(*did.MultiValidator)
+	validator.Validators = append(validator.Validators, managedServiceValidator{
+		serviceResolver: serviceResolver,
+	})
+	return validator
```

Comment:
you could just wrap it in another multivalidator, then you don't need casting (dont mind the formatting)?

```suggestion
	return &did.MultiValidator{Validators: []did.Validator{NetworkDocumentValidator(), managedServiceValidator{
		serviceResolver: serviceResolver,
	}}
```

Reply:
The idea is that ManagedDocumentValidator does everything the NetworkDocumentValidator + some more. Extending the NetworkDocumentValidator guarantees that this is the case.